src/ledger/transactions.js

```diff
@@ -47,9 +47,6 @@ function parseAccountTxTransaction(tx) {
 }
 
 function counterpartyFilter(filters, tx: TransactionType) {
-  if (!filters.counterparty) {
-    return true;
```

Comment:
That check is done before this function called.
https://github.com/ripple/ripple-lib/blob/develop/src/ledger/transactions.js#L79